### PR TITLE
:sparkles: Simple classification models zoo built up

### PR DESCRIPTION
**Pull request by alexwwang**

### CODE REVIEW

**Comment by alexwwang** [comment]
@BrikerMan Hi, would you please check the CI check result and test codes related to errors raised?
I doubt if there's a bug induced in dealing with multi-tag, which caused these dimension errors.

I think these errors do none of business with model structure definitions.

**Comment by BrikerMan** [comment]
Sure, I will check it.

**Comment by alexwwang** [comment]
> Sure, I will check it.

Thanks in advance.

And behaviors respected to two python editions seem to be different in `preprocess` test, raising error in py3.6 but not in py3.7.

**Comment by BrikerMan** [comment]
Let me merge this pull request, then test~

**Comment by BrikerMan** [comment]
I have added several test_cases, I think these models have some structural error.

```
tests/classification/test_avrnn_model.py
tests/classification/test_dropout_avrnn.py
tests/classification/test_r_cnn.py
```

They should at least pass the `test_basic_use` test function. You could run those tests locally in Pycharm or VSCode. Or just run by command 
```bash
nosetests --with-coverage --cover-html --cover-html-dir=htmlcov \
    --cover-xml --cover-xml-file=coverage.xml --with-xunit --cover-package="kashgari" \ 
    $TEST_FILE
```

**Comment by BrikerMan** [comment]
I will check the `preprocess` error.

**Comment by alexwwang** [comment]
> Let me merge this pull request, then test~

OK.

I noticed that a list index out of range error raised in `KMax_CNN_Model` when attempting to concatenate a list of 4 tensors on axis 1. It's supposed to be no problem in anyhow if these tensors are 2D, say (batch_size, flattened_1d_vec). So I doubt the tensor passed to this layer is failed to maintain this shape or the error won't happen.

**Comment by alexwwang** [comment]
> I will check the `preprocess` error.

The error info in CI hinted that the type of var passing into a reshape function is `tuple`, which is supposed not to be.

**Comment by BrikerMan** [comment]
@alexwwang It is wired, this build runs successfully, [Build #460](https://travis-ci.com/BrikerMan/Kashgari/builds/117076579)

**Comment by alexwwang** [comment]
> @alexwwang It is wired, this build runs successfully, [Build #460](https://travis-ci.com/BrikerMan/Kashgari/builds/117076579)

And errors thrown out each time are not the same in classification test tasks.
In Build #460, KMax_CNN test passed successfully.
I've no idea what's happening but I decide to try these tests locally first.

**Comment by alexwwang** [comment]
Two customized layers may be not correctly implemented.

**Comment by alexwwang** [comment]
The discrepancies between Keras and tf.keras are much more than I assumed...
😂
It would take a longer time to fix these problems to pass the test than expected.

**Comment by BrikerMan** [comment]
> The discrepancies between Keras and tf.keras are much more than I assumed...
> 😂
> It would take a longer time to fix these problems to pass the test than expected.

We could release tf.keras version with basic models first. Then add those models with custom layers. I have tried to use CRF layer with tf.keras, needs to change a lot.

**Comment by alexwwang** [comment]
Even the model with concatenate structure needs to modify. But I haven't
figured out the reason why it raises errors yet.

**Comment by alexwwang** [comment]
@BrikerMan I have a problem, where is the implementation of `get_default_hyper_parameters` definition? And where could I find an entire sample of hyper_parameters dict?
I noticed a test task to test variable length model, I need to figure out how it works and if it is suitable to these models.

**Comment by BrikerMan** [comment]
@alexwwang `get_default_hyper_parameters` defined inside each model. It should return a dict. I think you will have to write a for loop to get all sample of hyper_parameters dict. If a test function will fail in a model class, we could rewrite the test case for a specific model. Lets' assume `test_custom_hyper_params` function fail on `CNN_GRU_Model`. Then change `test_cnn_gru.py` file to this.

PS. the test name `test_variable_length_model` is misguiding, changed to `test_custom_hyper_params`.

```
class TestBiGRUModel(base.TestBi_LSTM_Model):
    @classmethod
    def setUpClass(cls):
        cls.epochs = 1
        cls.model_class = CNN_GRU_Model

    def test_custom_hyper_params(self):
        # do it the right way
```

**Comment by alexwwang** [comment]
then I guess I know why.

**Comment by BrikerMan** [comment]
> then I guess I know why.
> add 15 to a value, then list assignment index out of range... 😂
> 
> I mistakenly thought this aims to test model with input data of variable length... 🤦‍♂️

Sorry for the misleading name. 🤦‍♂️

**Comment by alexwwang** [comment]
Never mind.
It doesn't cost too much. 😁
